gateway-ha/src/main/java/io/trino/gateway/ha/handler/RoutingTargetHandler.java

```diff
@@ -45,18 +46,24 @@ public class RoutingTargetHandler
     private final RoutingGroupSelector routingGroupSelector;
     private final List<String> statementPaths;
     private final List<Pattern> extraWhitelistPaths;
+    private final boolean requestAnalyserClientsUseV2Format;
+    private final int requestAnalyserMaxBodySize;
     private final boolean cookiesEnabled;
 
     public RoutingTargetHandler(
             RoutingManager routingManager,
             RoutingGroupSelector routingGroupSelector,
             List<String> statementPaths,
-            List<String> extraWhitelistPaths)
+            List<String> extraWhitelistPaths,
+            RequestAnalyzerConfig requestAnalyzerConfig)
```

Comment:
I think we should bind this class and inject `RoutingManager` & `RoutingGroupSelector` instead of initializing in `HaGatewayProviderModule#getRoutingTargetHandler`.

Reply:
should that be a separate PR, or should I wrap it into this one?

Reply:
Follow-up PR is fine for me.

Reply:
https://github.com/trinodb/trino-gateway/pull/488 a separate PR seemed simpler